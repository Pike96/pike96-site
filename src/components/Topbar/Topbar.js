import React from 'react';
import styles from './Topbar.module.scss';
import { useSiteMetadata } from '../../hooks';
import { getRandomItemFromStrArray } from '../../utils';

const Topbar = () => {
  const { keywords } = useSiteMetadata();

  return (
    <div className={styles['topbar']}>
      <div className={styles['topbar__keyword']}>
        &lt;pike96&gt;
        <span className={styles['topbar__keyword-inner']}>
          {getRandomItemFromStrArray(keywords)}
        </span>
        &lt;/pike96&gt;
      </div>
    </div>
  );
};

export default Topbar;