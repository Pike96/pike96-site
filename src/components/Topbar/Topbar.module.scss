@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.topbar {
  position: fixed;
  height: 2.5rem;
  width: 100%;
  top: 0;
  left: 0;
  background: $color-primary;

  &__keyword {
    margin: auto;
    color: $color-white;
    font-family: monospace;
    font-size: 22px;
    height: $topbar-height;
    line-height: $topbar-height;
    background-color: $color-primary;
    text-align: center;
  }
}