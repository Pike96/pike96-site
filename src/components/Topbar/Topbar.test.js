// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Topbar from './Topbar';

describe('Topbar', () => {
  const props = {
    children: 'test',
    title: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Topbar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
