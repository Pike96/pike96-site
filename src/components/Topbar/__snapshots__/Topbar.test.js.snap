// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar renders correctly 1`] = `
<div
  className="topbar"
>
  <div
    className="topbar__keyword"
  >
    &lt;pike96&gt;
    <span
      className="topbar__keyword-inner"
    >
      Test keyword 3.
    </span>
    &lt;/pike96&gt;
  </div>
</div>
`;
